packages/cactus-api-client/src/main/typescript/socketio-api-client.ts

```diff
@@ -0,0 +1,345 @@
+    } catch (err) {
+      this.log.error("sendAsyncRequest() EXCEPTION", err);
+      throw err;
+    }
+  }
+
+  // TODO - Refactor to RxJS
+  public sendSyncRequest(
+    contract: Record<string, unknown>,
+    method: Record<string, unknown>,
+    args: any,
+  ): Promise<any> {
+    return new Promise((resolve, reject) => {
+      this.log.debug("call : sendSyncRequest");
+
+      try {
+        this.log.debug(
+          "##in sendSyncRequest, contract:",
+          contract,
+          "method:",
+          method,
+          "args:",
+          args,
+        );
+        let responseFlag = false;
+
+        // reqID generation
+        const reqID = this.genarateReqID();
+        this.log.debug(`##sendSyncRequest, reqID = ${reqID}`);
+
+        this.socket.on("connect_error", (err: Error) => {
+          this.log.error("##connect_error:", err);
+          this.socket.disconnect();
+          reject(err);
+        });
+        this.socket.on("connect_timeout", (err: Record<string, unknown>) => {
+          this.log.error("####Error:", err);
+          this.socket.disconnect();
+          reject(err);
+        });
+        this.socket.on("error", (err: Record<string, unknown>) => {
+          this.log.error("####Error:", err);
+          this.socket.disconnect();
+          reject(err);
+        });
+        this.socket.on("response", (result: any) => {
+          this.log.debug("#[recv]response, res:", result);
+          if (reqID === result.id) {
+            responseFlag = true;
+
+            this.checkValidator(
+              this.options.validatorKeyPath,
+              result.resObj.data,
+            )
+              .then((decodedData) => {
+                this.log.debug("checkValidator decodedData:", decodedData);
+                const resultObj = {
+                  status: result.resObj.status,
+                  data: decodedData.result,
+                };
+                this.log.debug("resultObj =", resultObj);
+                // Result reply
+                resolve(resultObj);
+              })
+              .catch((err) => {
+                responseFlag = false;
+                this.log.debug("checkValidator error:", err);
+                this.log.error(err);
+              });
+          }
+        });
+
+        // Call Validator
+        const requestData = {
+          contract: contract,
+          method: method,
+          args: args,
+          reqID: reqID,
+        };
+        this.log.debug("requestData:", requestData);
+        this.socket.emit("request2", requestData);
+        this.log.debug("set timeout");
+
+        // Time-out setting
+        const timeoutMilliseconds =
+          this.options.syncFunctionTimeoutMillisecond ||
+          defaultSyncFunctionTimeoutMillisecond;
+        setTimeout(() => {
+          if (responseFlag === false) {
+            this.log.debug("requestTimeout reqID:", reqID);
+            resolve({ status: 504, amount: 0 });
```

Comment:
@outSH What does `amount` refer to here? Is that something specific to a certain use-case?

Reply:
@petermetz I consulted japanese colleagues and it seems this amount field is obsolete. I've removed this in most recent version.